v13 types incorrectly points to API Version "2022-11-15"

### Describe the bug

v13 types incorrectly pins to "2022-11-15", it should pin to "2023-08-16" as mentioned in the [release notes](https://github.com/stripe/stripe-node/releases/tag/v13.0.0).

https://github.com/stripe/stripe-node/blob/3065b53f868475eee1147361b0710371df5240c3/types/lib.d.ts#L30

It gives error when trying to use api version "2023-08-16" with `new Stripe(<API_KEY>, { apiVersion: "2023-08-16" })`

```ts
src/stripe.ts:7:3 - error TS2322: Type '"2023-08-16"' is not assignable to type '"2022-11-15"'.

7   apiVersion: '2023-08-16',
    ~~~~~~~~~~
```

### To Reproduce

With `stripe@13.0.0` try running this code

```ts
const stripe = new Stripe(<API_KEY>, {
  apiVersion: "2023-08-16"
});
```

### Expected behavior

No type error

### Code snippets

_No response_

### OS

Arch Linux

### Node version

Node v20.2.0

### Library version

stripe-node v13.0.0

### API version

2023-08-16

### Additional context

_No response_

--- a/examples/webhook-signing/express/main.ts
+++ b/examples/webhook-signing/express/main.ts
@@ -8,7 +8,7 @@ import {AddressInfo} from 'net';
 env.config();
 
 const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
-  apiVersion: '2022-11-15',
+  apiVersion: '2023-08-16',
 });
 
 const webhookSecret: string = process.env.STRIPE_WEBHOOK_SECRET;

--- a/examples/webhook-signing/koa/main.ts
+++ b/examples/webhook-signing/koa/main.ts
@@ -12,7 +12,7 @@ env.config();
 const webhookSecret = process.env.STRIPE_WEBHOOK_SECRET;
 
 const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
-  apiVersion: '2022-11-15',
+  apiVersion: '2023-08-16',
 });
 
 const handleWebhook = async (ctx: Koa.ParameterizedContext, next: Koa.Next) => {

--- a/examples/webhook-signing/nextjs/pages/api/webhooks.ts
+++ b/examples/webhook-signing/nextjs/pages/api/webhooks.ts
@@ -6,7 +6,7 @@ const handler = async (
   res: NextApiResponse
 ): Promise<void> => {
   const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
-    apiVersion: '2022-11-15',
+    apiVersion: '2023-08-16',
   });
 
   const webhookSecret: string = process.env.STRIPE_WEBHOOK_SECRET;

--- a/examples/webhook-signing/test/main.ts
+++ b/examples/webhook-signing/test/main.ts
@@ -42,7 +42,7 @@ env.config({
 
 const webhookSecret = process.env.STRIPE_WEBHOOK_SECRET;
 const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
-  apiVersion: '2022-11-15',
+  apiVersion: '2023-08-16',
 });
 
 const payload = Buffer.from(

--- a/testProjects/cloudflare-pages/functions/index.js
+++ b/testProjects/cloudflare-pages/functions/index.js
@@ -5,7 +5,7 @@ export async function onRequestPost({env, request}) {
   const body = await request.text();
 
   const stripe = new Stripe(env.STRIPE_API_KEY, {
-    apiVersion: '2022-11-15',
+    apiVersion: '2023-08-16',
     httpClient: Stripe.createFetchHttpClient(),
   });
   const webCrypto = Stripe.createSubtleCryptoProvider();

--- a/testProjects/mjs-ts/index.ts
+++ b/testProjects/mjs-ts/index.ts
@@ -1,13 +1,13 @@
 import DefaultStripe, {Stripe} from 'stripe';
 
 const stripe = new Stripe(process.argv[2], {
-  apiVersion: '2022-11-15', 
+  apiVersion: '2023-08-16',
   host: process.env.STRIPE_MOCK_HOST || 'localhost',
   port: process.env.STRIPE_MOCK_PORT || 12111,
   protocol: 'http',
 });
 const defaultStripe = new DefaultStripe(process.argv[2], {
-  apiVersion: '2022-11-15',
+  apiVersion: '2023-08-16',
   host: process.env.STRIPE_MOCK_HOST || 'localhost',
   port: process.env.STRIPE_MOCK_PORT || 12111,
   protocol: 'http',

--- a/types/lib.d.ts
+++ b/types/lib.d.ts
@@ -27,7 +27,7 @@ declare module 'stripe' {
       }): (...args: any[]) => Response<ResponseObject>; //eslint-disable-line @typescript-eslint/no-explicit-any
       static MAX_BUFFERED_REQUEST_METRICS: number;
     }
-    export type LatestApiVersion = '2022-11-15';
+    export type LatestApiVersion = '2023-08-16';
     export type HttpAgent = Agent;
     export type HttpProtocol = 'http' | 'https';
 

--- a/types/test/typescriptTest.ts
+++ b/types/test/typescriptTest.ts
@@ -27,7 +27,7 @@ stripe = new Stripe('sk_test_123', {
 
 // Check config object.
 stripe = new Stripe('sk_test_123', {
-  apiVersion: '2022-11-15',
+  apiVersion: '2023-08-16',
   typescript: true,
   maxNetworkRetries: 1,
   timeout: 1000,

--- a/types/test/typescriptTest.ts
+++ b/types/test/typescriptTest.ts
@@ -9,7 +9,7 @@
 import Stripe from 'stripe';
 
 let stripe = new Stripe('sk_test_123', {
-  apiVersion: '2022-11-15',
+  apiVersion: '2023-08-16',
 });
 
 // @ts-ignore lazily ignore apiVersion requirement.

